compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -364,6 +365,50 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                             }
                         })
                         .collect()
+                } else if let Rest::DefaultFields(field_types) = base {
+                    // // `field_names` is an iterator, ok to clone.
+                    // for (n, ty) in iter::zip(field_names.clone(), &**field_types) {
+                    //     if let None = fields_map.get(&n)
+                    //         && let None = variant.fields[n].value
+                    //     {
+                    //         this.tcx
+                    //             .dcx()
+                    //             .struct_span_err(expr_span, format!("asdf {n:?} {ty}"))
+                    //             .emit();
+                    //         return block.unit();
+                    //     }
+                    // }
+                    iter::zip(field_names, &**field_types)
+                        .map(|(n, ty)| match fields_map.get(&n) {
+                            Some(v) => v.clone(),
+                            None => {
+                                match variant.fields[n].value {
+                                    Some(value) => this.literal_operand(
+                                        expr_span,
+                                        Const::from_ty_const(value, *ty, this.tcx)
+                                            .normalize(this.tcx, this.param_env),
+                                    ),
+                                    None => {
+                                        let name = variant.fields[n].name;
+                                        this.tcx
+                                            .dcx()
+                                            .struct_span_err(
+                                                expr_span,
+                                                format!(
+                                                    "missing mandatory field `{name}` of type \
+                                                     `{ty}`",
+                                                ),
+                                            )
+                                            .emit();
+                                        // FIXME: uncomment the prior check and fall-back in a more
+                                        // graceful manner.
+                                        let place = this.temp(*ty, expr_span);
+                                        this.consume_by_copy_or_move(place)
```

Comment:
This is the only thing about this implementation that is iffy, we need a better approach for this, particularly to avoid the knock-down error stating that the value is uninitialized.

Reply:
You can taint the mir body, so the compiler does not try to borrowck it.